File: July-LeetCoding-Challenge/25-Find-Minimum-in-Rotated-Sorted-Array-II/Find-Minimum-in-Rotated-Sorted-Array-II.cs
public class Solution {
    public int FindMin(int[] nums) {
        var h = nums.Length-1;
        var l = 0;
        while(l < h){
            int mid = (l + h-1)/2;
            if(nums[mid] > nums[h]){
                l = mid +1;
            }else if(nums[mid] < nums[l]){
                h = mid;
                l++;
            }else {
                h--;
            }
        }
        return nums[l];
    }   
}

File: July-LeetCoding-Challenge/25-Find-Minimum-in-Rotated-Sorted-Array-II/Find-Minimum-in-Rotated-Sorted-Array-II.kt
package binary_search

class FindMinimuminRotatedSortedArrayKotlin153 {
    fun findMin(nums: IntArray): Int {
        var left = 0
        var right = nums.size - 1
        while (left + 1 < right) {
            val mid = left + (right - left) / 2
            when {
                nums[mid] >= nums[0] -> {
                    if (nums[mid] < nums[nums.size - 1]) {
                        return nums[0]
                    } else {
                        left = mid
                    }
                }
                nums[mid] <= nums[nums.size - 1] -> {
                    right = mid
                }
            }
        }
        return minOf(nums[left], nums[right])
    }
}